packages/picasso.js/src/core/data/extractor.js

```diff
@@ -42,7 +42,12 @@ export default function extract(dataConfig, data = {}, opts = {}) {
           if (!s) {
             return;
           }
-          extracted.items.push(...s.extract(cfg));
+          const tmp = s.extract(cfg);
+          if (tmp.length < 100000) {
+            extracted.items.push(...tmp);
+          } else {
```

Comment:
Any purpose to keep the if-else, instead of always going with the else case?

Reply:
If we push in the same array then the memory will be less than change to a new array I guess.

Reply:
Regarding my comment above, if we simple just use the spread syntax for all cases the need for that arbitrary number disappears.

It might be slower, but that is going to be a big might and may very on different browsers.

On top of that it should also be safer to just use [...ary1, ...ary2].

Reply:
Updated by not using push.